dags/iop/parser.py

```diff
@@ -106,6 +117,51 @@ def _get_extracted_arxiv_eprint_value(self, article):
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
 
+    def _get_date_element(self, article):
+        try:
+            date = (
+                article.find("front/article-meta/history/date[@date-type='published']")
+                or article.find("front/article-meta/history/date[@date-type='epub']")
+                or article.find("front/article-meta/history/pub-date[@pub-type='ppub']")
+                or article.find("front/article-meta/pub-date")
+            )
+            return date
+        except AttributeError:
+            return
+
+    def _get_date_published(self, article):
+        date_element = self._get_date_element(article)
+        return self._get_date(date_element)
+
+    def _get_date(self, date):
+        day = None
+        month = None
```

Comment:
why do you need to define them here? can't you assign them to `None` in `except` statement?

Reply:
Actually, I can